modules/node.js

```diff
@@ -68,15 +68,20 @@ Node.prototype.internal = {
 	getForgingStatus(publicKey, cb) {
 		var keyPairs = modules.delegates.getForgersKeyPairs();
 		var internalForgers = library.config.forging.secret;
+		var forgersPublicKeys = {};
+
+		for (var pair in keyPairs) {
+			forgersPublicKeys[keyPairs[pair].publicKey.toString('hex')] = true;
+		}
 
 		var fullList = internalForgers.map(forger => ({
-			forging: !!forger.publicKey,
+			forging: !!forgersPublicKeys[forger.publicKey],
```

Comment:
Why new variable instead of just `!!keyPairs[forger.publicKey]`?

Reply:
Because `keyPairs` stores key pairs as object index, not publicKey, therefore the expression `keyPairs[forger.publicKey]` is not valid. The right index would be `keyPairs[keyPair]`

Reply:
No? It's created like that:
`__private.keypairs[keypair.publicKey.toString('hex')] = keypair;`
Also we have a test that confirms that behavior.

Reply:
Yes!, good catch Mariusz